src/data/zmod/units.lean

```diff
@@ -0,0 +1,174 @@
+/-
+Copyright (c) 2023 Ashvni Narayanan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ashvni Narayanan
+-/
+import data.zmod.basic
+import topology.algebra.constructions
+
+/-!
+# Properties of units of ℤ/nℤ
+
+This file enlists some properties of units of `zmod n`. We then define a topological structure
+(the discrete topology) on `zmod n` for all `n` and consequently on the units.
+We also enlist several properties that are helpful with modular arithmetic.
+
+## Main definitions and theorems
+ * `zmod.topological_space`
+ * `cast_hom_apply'` : a version of `cast_hom_apply` where `R` is explicit
+ * `disc_top_units` : giving discrete topology to `units (zmod n)`
+
+## Tags
+zmod, units
+-/
+
+open zmod nat
+namespace zmod
+/-- Same as `zmod.cast_hom_apply` with some hypotheses being made explicit. -/
+lemma cast_hom_apply' {n : ℕ} (R : Type*) [ring R] {m : ℕ} [char_p R m]
+  (h : m ∣ n) (i : zmod n) : (cast_hom h R) i = ↑i := cast_hom_apply i
+
+lemma coe_map_of_dvd {a b : ℕ} (h : a ∣ b) (x : units (zmod b)) :
+  is_unit (x : zmod a) :=
```

Comment:
Either have a lemma `is_unit _ → is_unit _` or a function `(zmod b)ˣ → (zmod a)ˣ` (possibly we want both!), but not a mix of them like this.

Reply:
Its true that `a ∣ b → (zmod b → zmod a)`? I think this follows trivially from that.

Reply:
It is a single line proof, yes. Shall I remove it? I possibly need it for future work but I can add it then.

Reply:
It's true that `a ∣ b → (zmod b →+* zmod a)`. It wouldn't surprise me if mathlib already had a map `zmod b -> zmod a` for arbitrary a and b :-/